tests/test_platform.rb

```diff
@@ -221,6 +221,10 @@ def test_fans
   end
 
   def test_virtual_services
+    # Only run this test if a virtual-service is installed
+    if @device.cmd('show virtual-service global')[/services installed : 0$/]
```

Comment:
1. I guess... it's only excluded on ios_xr... seems like a waste of valuable coding space but will add it.
2. Sure.  Don't have a strong preference either way.